File: Problem-038/solution.js
function pandigitalMultiples(k) {
    let largest = 0;
    const target = "";
    let required = "";
    for (let i = 1; i <= k; i++) {
        required += i;
    }
    for (let num = 1; num < 100000; num++) {
        let combined = "";
        let multiplier = 1;
        while (combined.length < k) {
            combined += num * multiplier;
            multiplier++;
        }
        if (
            combined.length === k &&
            multiplier > 2 &&
            isPandigital(combined, required)
        ) {
            largest = Math.max(largest, Number(combined));
        }
    }

    return largest;
}


function isPandigital(str, required) {
    return str.split("").sort().join("") === required;
}

File: Problem-038/solution.py
def pandigital_multiples(k):
    largest = 0

    # Required digits, e.g. k = 5 → "12345"
    required = "".join(str(i) for i in range(1, k + 1))

    # Try different starting numbers
    for num in range(1, 100000):
        combined = ""
        multiplier = 1

        # Concatenate num×1, num×2, num×3...
        while len(combined) < k:
            combined += str(num * multiplier)
            multiplier += 1

        if (
            len(combined) == k
            and multiplier > 2
            and "".join(sorted(combined)) == required
        ):
            largest = max(largest, int(combined))

    return largest


print(pandigital_multiples(9))
